### PR TITLE
[CALCITE-3220] HiveSqlDialect transforms function Trim to correct Hive format

### PR DESCRIPTION
**Pull request by JackyWoo**
Let's  assume sql = SELECT TRIM(' str ')

When we use HiveSqlDialect and transform sql, we expect  **SELECT TRIM(' str ')**，but get  **SELECT TRIM(BOTH ' ' FROM ' str ')**.

Hive function trim, ltrim, rtrim usage see [hive UDF usage](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF)

This pr change HiveSqlDialect behavior when transform function trim.

### CODE REVIEW

**Comment by hsyuan** [comment]
Any JIRA for this? Please link with JIRA

**Comment by JackyWoo** [comment]
hi, hsyuan

jira  CALCITE-3220

File: core/src/test/java/org/apache/calcite/sql/dilect/HiveSqlDialectTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (DonnyZone) [inline core/src/test/java/org/apache/calcite/sql/dilect/HiveSqlDialectTest.java]: You can test the function in RelToSqlConverterTest. A new HiveSqlDialectTest is not necessary.

**Comment by DonnyZone** [comment]
Maybe you can put a reference link for the Trim function usage in Hive.

File: core/src/test/java/org/apache/calcite/sql/dilect/HiveSqlDialectTest.java (hunk 1 above)
Comment (JackyWoo) [reply core/src/test/java/org/apache/calcite/sql/dilect/HiveSqlDialectTest.java]: Thanks for advise. 
I have put test in RelToSqlConverterTest.

**Comment by JackyWoo** [comment]
@DonnyZone ok, Thanks for advice.

**Pull request closed without merging**